gradoop-flink/src/main/java/org/gradoop/flink/io/impl/text/GDLEncoder.java

```diff
@@ -0,0 +1,254 @@
+   */
+  private static final String VERTEX_VARIABLE_PREFIX = "v";
+  /**
+   * Seperates properties.
+   */
+  private static final String PROPERTIES_PREFIX = "{";
+  /**
+   * Seperates properties.
+   */
+  private static final String PROPERTIES_SUFFIX = "}";
+  /**
+   * Seperates properties.
+   */
+  private static final String PROPERTIES_SEPERATOR = ",";
+  /**
+   * Seperates key and value for properties.
+   */
+  private static final String KEY_VALUE_SEPERATOR = ":";
+  /**
+   * Suffix for GDL double representation.
+   */
+  private static final String DOUBLE_SUFFIX = "d";
+  /**
+   * Suffix for GDL float representation.
+   */
+  private static final String FLOAT_SUFFIX = "f";
+  /**
+   * Suffix for GDL long representation.
+   */
+  private static final String LONG_SUFFIX = "L";
+  /**
+   * GDL null representation.
+   */
+  private static final String NULL_STRING = "NULL";
+  /**
+   * GDL string prefix
+   */
+  private static final String STRING_PREFIX = "\"";
+  /**
+   * GDL string suffix
+   */
+  private static final String STRING_SUFFIX = "\"";
+
+  /**
+   * Graph head to encode.
+   */
+  private GraphHead graphHead;
+  /**
+   * Vertices to encode.
+   */
+  private List<Vertex> vertices;
+  /**
+   * Edges to encode.
+   */
+  private List<Edge> edges;
+
+  /**
+   *
+   * @param graphHead graph head that should be encoded
+   * @param vertices vertices that should be encoded
+   * @param edges edges that should be encoded
+   */
+  public GDLEncoder(GraphHead graphHead, List<Vertex> vertices, List<Edge> edges) {
+    this.graphHead = graphHead;
+    this.vertices = vertices;
+    this.edges = edges;
+  }
+
+  /**
+   * Transforms the graph that is represented by graph head, vertices and edges into a GDL string.
+   * @return GDL formatted String.
+   */
+  public String graphToGDLString() {
+
+    Map<GradoopId, String> idToVertexName = getVertexNameMapping(vertices);
+
+
+    String graphHeadString = graphHeadToGDLString(graphHead);
+
+    String verticesString = vertices.stream()
+      .map(v -> vertexToGDLString(v, idToVertexName.get(v.getId())))
+      .collect(Collectors.joining("\n"));
+
+    String edgesString = edges.stream()
+      .map(e -> edgeToGDLString(e, idToVertexName))
+      .collect(Collectors.joining("\n"));
+
+    StringBuilder result = new StringBuilder()
+      .append(graphHeadString)
+      .append(GRAPH_ELEMENTS_DEFINITION_START).append("\n")
+      .append(verticesString).append("\n")
+      .append(edgesString).append("\n")
+      .append(GRAPH_ELEMENTS_DEFINITION_END);
+
+    return result.toString();
+  }
+
+  /**
+   * Returns a mapping between the vertex gradoop id and the gdl variable name.
+   * @param vertices The graph vertices.
+   * @return Mapping between vertex and gdl variable name.
+   */
+  private Map<GradoopId, String> getVertexNameMapping(List<Vertex> vertices) {
+    Map<GradoopId, String> idToVertexName = new HashMap<>();
```

Comment:
since you add every vertex to the new map you can create with expected size
new HashMap<>(vertices.size);